### ISSUE TITLE
HAWQ: Hessian AWare Quantization of Neural Networks With Mixed-Precision

### ISSUE THREAD

**Issue by joapolarbear** [joapolarbear/dl_notes#13]
### ICCV 2019
[PDF](https://openaccess.thecvf.com/content_ICCV_2019/html/Dong_HAWQ_Hessian_AWare_Quantization_of_Neural_Networks_With_Mixed-Precision_ICCV_2019_paper.html)

### Automatically select the relative quantization precision of each layer, i.e., use fp16, int8 or one bit
> 1. not all layers have the same distribution of floating-point values
> 2. the network can have significantly different **sensitive** to the quantization of each layer.

* One-order gradients are not enough to decide the sensitive, so use second-order Hessian matrix (maximum eigenvalue *\lambda*) to decide the **sensitive**, furthermore, considering the size of a layer, (denoted as *n*), **sensitive** of one layer = *\lambda* / *n*.
* Use the pre-trained network to calculate Hessian matrix, 
* **matrix-free power iteration algorithm**: avoid explicitly form the Hessian due to the large size.

### Multi-stage fine-tuning -- define a fine-tuning order
re-train, sort layers in descending order according to the product of \lambda and weight difference.

**Comment by joapolarbear** [comment]
![image](https://user-images.githubusercontent.com/17765864/86873130-4c2ee280-c110-11ea-93b7-8718e108ebbe.png)
